src/syzygy/tbcore.cpp

```diff
@@ -133,125 +144,121 @@ static void unmap_file(char *data, uint64_t mapping)
 }
 #endif
 
-static void add_to_hash(TBEntry *ptr, uint64_t key)
+static void add_to_hash(TBEntry* ptr, uint64_t key)
 {
-    int i, hshidx;
-
-    hshidx = key >> (64 - TBHASHBITS);
-    i = 0;
+    TBHashEntry* entry = TB_hash[key >> (64 - TBHASHBITS)];
 
-    while (i < HSHMAX && TB_hash[hshidx][i].ptr)
-        i++;
+    for (int i = 0; i < HSHMAX && entry->ptr; i++, entry++) {}
 
-    if (i == HSHMAX) {
+    if (!entry->ptr) {
+        entry->key = key;
+        entry->ptr = ptr;
+    } else {
         std::cerr << "HSHMAX too low!\n";
         exit(1);
-    } else {
-        TB_hash[hshidx][i].key = key;
-        TB_hash[hshidx][i].ptr = ptr;
     }
 }
 
 static const std::string pchr = " PNBRQK";
 
-static void init_tb(const std::string& str)
+static void init_tb(const std::vector<PieceType>& pieces)
 {
     FD fd;
-    TBEntry *entry;
-    int i, j, pcs[16];
-    uint64_t key, key2;
-    int color;
-
-    fd = open_tb(str, WDLSUFFIX);
+    TBEntry* entry;
+    uint64_t key1, key2;
 
-    if (fd == FD_ERR)
-        return;
+    std::string fname;
+    Color c = BLACK;
+    uint8_t pcs[PIECE_NB] = {0};
+    int num = 0;
 
-    close_tb(fd);
+    for (PieceType pt : pieces)
+    {
```

Comment:
Ok I will although please note that SF standard is different (and more
readable IMO).

In this case, that is a kind of 3party library we can have a different code
style, and I will adapt to the one you choose because you are the leader of
this sub-project, but again, SF style is more clear, especially when
dealing with complex code like this one.



On Tue, Apr 19, 2016 at 1:35 PM, lucasart <notifications@github.com> wrote:

> In src/syzygy/tbcore.cpp
> <https://github.com/official-stockfish/Stockfish/pull/641#discussion_r60215023>
> :
>
> >
> > -    close_tb(fd);
> > +    for (PieceType pt : pieces)
> > +    {
>
> Is it possible to retain K&R style of braces for now ? I used astyle -A3
> -s4 -f -xn -xc -xl, and try to stick to it for now, so I never need to
> override it manually. Eventually, when the code is mature, we can switch to
> SF style (which is very awkward btw), but better do it with an
> auto-indenter, all in one go.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/official-stockfish/Stockfish/pull/641/files/decb8b383f40787aab2522f2716bca6c7fdc342d#r60215023>
>